### PR TITLE
Configure Dockerfile for staging environment

### LINKED ISSUE #191: Prepare the new frontend for staging environment
**Is your feature request related to a problem? Please describe.**
Once we're done w/ migration to the new frontend we'll need to run it within our [docker-compose-staging.yaml ](https://github.com/OWASP/Nest/blob/0629eba2aec1e1742431cde3e155cebaa32cafa4/docker-compose-staging.yaml) stack. We currently have some content in the [Dockerfile.staging](https://github.com/OWASP/Nest/blob/0629eba2aec1e1742431cde3e155cebaa32cafa4/frontend/Dockerfile.staging) but it has never been tested/used.

**Describe the solution you'd like**
Revisit Dockerfile.staging for the new frontend and modify/improve if needed. Update docker-compose-staging.yaml to include the new frontend in staging environment

### PR DESCRIPTION
**Pull request by harsh3dev**
Resolves #191 

This pull request includes several changes to the `docker-compose-staging.yaml`, `frontend/Dockerfile.staging`, and `nginx/staging.conf` files to integrate and serve the frontend application in the staging environment. The most important changes are listed below:

Docker Compose configuration updates:
- [x] Added a new `frontend` service configuration to the `docker-compose-staging.yaml` file. This includes setting up the container, build context, environment variables, and networking.
- [x] Updated the `nginx` service configuration to use `staging.conf` instead of `nginx.conf` and added dependency on the `frontend` service.
- [ ] Commit the requested changes and test in local env

### CODE REVIEW

File: frontend/Dockerfile.staging [hunk 1]
```diff
@@ -7,6 +7,7 @@ RUN npm install
 
 COPY . .
 RUN npm run build
+COPY .env .env
```
Comment (arkid15r) [inline frontend/Dockerfile.staging:10]: This isn't right. In this case you'll have your secrets in the image which is available for downloading to anyone.

File: docker-compose-staging.yaml [hunk 2]
```diff
@@ -21,6 +21,18 @@ services:
     volumes:
       - ./data:/home/owasp/data
 
+  frontend:
+    container_name: nest-frontend
+    build:
+      context: ./frontend
+      dockerfile: Dockerfile.staging
+    env_file: ./frontend/.env
```
Comment (arkid15r) [inline docker-compose-staging.yaml:29]: Use backend naming approach here too

File: docker-compose-staging.yaml [hunk 3]
```diff
@@ -21,6 +21,18 @@ services:
     volumes:
       - ./data:/home/owasp/data
 
+  frontend:
+    container_name: nest-frontend
+    build:
```
Comment (arkid15r) [inline docker-compose-staging.yaml:26]: We don't build on staging, we want to use a prebuilt image (see how it's done for backend)

File: docker-compose-staging.yaml [hunk 4]
```diff
@@ -21,6 +21,18 @@ services:
...
+      context: ./frontend
+      dockerfile: Dockerfile.staging
+    env_file: ./frontend/.env
+    restart: unless-stopped
+    networks:
+      - nest_app_network
+    expose:
+      - "80"
```
Comment (arkid15r) [inline docker-compose-staging.yaml:34]: Let's use 3000 for the FE for consistency.

**Review by arkid15r — CHANGES_REQUESTED** [review]
This needs some more work:

**Comment by yashpandey06** [comment]
@arkid15r  can i work on this ?...the image will be served from docker-hub for frontend.

**Comment by arkid15r** [comment]
Implemented in https://github.com/OWASP/Nest/commit/31626655b0cb5ea68ccbcc9f4bbb874496c6b83c

**Pull request closed without merging**